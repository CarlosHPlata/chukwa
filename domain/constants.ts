export const CURRENCY_PRECISION = 1000; // 1 euro = 1000 cents
export const VIEW_DATETIME_FORMAT = "dd-MM-yyyy HH:mm";
export const VIEW_DATE_FORMAT = "dd-MM-yyyy";
export const SAVE_DATE_FORMAT = "yyyy-MM-dd'T'HH:mm:ss.SSSZZ"; // ISO 8601 format for Luxon
