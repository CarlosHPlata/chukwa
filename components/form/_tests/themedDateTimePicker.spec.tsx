import { VIEW_DATE_FORMAT, VIEW_DATETIME_FORMAT } from "@/domain/constants";
import { render } from "@testing-library/react-native";
import { DateTime } from "luxon";
import ThemedDateTimePicker from "../themedDateTimePicker";

describe("ThemedDateTimePicker", () => {
  it("renders correctly", () => {
    const date = DateTime.fromISO("2023-10-01T12:00:00").toJSDate();
    const mockOnDateChanged = jest.fn();
    const tree = render(
      <ThemedDateTimePicker
        label="date"
        value={date}
        onDateChanged={mockOnDateChanged}
        mode="datetime"
      />,
    );

    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with initial date in date mode", () => {
    const initialDate = DateTime.fromISO("2023-10-01T12:00:00")
    const date = initialDate.toJSDate();
    const expectedFormattedDate = initialDate.toFormat(VIEW_DATE_FORMAT);
    const mockOnDateChanged = jest.fn();
    const { getByText } = render(
      <ThemedDateTimePicker
        label="date"
        value={date}
        onDateChanged={mockOnDateChanged}
        mode="date"
      />,
    );

    expect(getByText(expectedFormattedDate)).toBeTruthy();
  });
  
  it("renders correctly with initial date in datetime mode", () => {
    const initialDate = DateTime.fromISO("2023-10-01T12:00:00")
    const date = initialDate.toJSDate();
    const expectedFormattedDate = initialDate.toFormat(VIEW_DATETIME_FORMAT);

    const mockOnDateChanged = jest.fn();
    const { getByText } = render(
      <ThemedDateTimePicker
        label="date"
        value={date}
        onDateChanged={mockOnDateChanged}
        mode="datetime"
      />,
    );

    expect(getByText(expectedFormattedDate)).toBeTruthy();
  });
});
