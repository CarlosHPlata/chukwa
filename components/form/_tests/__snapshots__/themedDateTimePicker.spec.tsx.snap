// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemedDateTimePicker renders correctly 1`] = `
[
  <Text>
    date
  </Text>,
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text>
      01-10-2023 12:00
    </Text>
  </View>,
]
`;
